source/common/network/dns_impl.h

```diff
@@ -1,47 +1,68 @@
 #pragma once
 
+#include "envoy/event/file_event.h"
 #include "envoy/network/dns.h"
 
 #include "common/common/linked_object.h"
 #include "common/event/dispatcher_impl.h"
 
+#include "ares.h"
 #include "event2/event_struct.h"
 
 namespace Network {
 
+class DnsResolverImplPeer;
```

Comment:
I think I see that you roughly need to tweak the resolver behavior for the tests. Can we make the constructor take params? Or have some virtual function which a test resolver overrides?